### PR TITLE
feat: Add Support for ECR Registries

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

- Added table for `aws_ecr_registries`
- Added table for `aws_ecr_registry_policies`

### CODE REVIEW

**Comment by disq** [comment]
I think the `validateResource` method in `source_testing.go` needs an update, but not sure how.
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/ecr#GetRegistryPolicyOutput

File: plugins/source/aws/resources/services/ecr/registry_policies_fetch.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchEcrRegistryPolicies(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().ECR
+	output, err := svc.GetRegistryPolicy(ctx, &ecr.GetRegistryPolicyInput{})
```
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/resources/services/ecr/registry_policies_fetch.go:14]: https://github.com/cloudquery/cloudquery/issues/2631

**Review by hermanschaaf — CHANGES_REQUESTED** [review]

File: plugins/source/aws/codegen/recipes/ecr.go [hunk 2]
```diff
@@ -1,13 +1,57 @@
...
 	"github.com/cloudquery/plugin-sdk/schema"
 )
 
 func ECRResources() []*Resource {
 	resources := []*Resource{
+		{
+			SubService: "registries",
+			Struct:     &ecr.DescribeRegistryOutput{},
+			SkipFields: []string{"RegistryId", "ResultMetadata"},
+			Multiplex:  `client.AccountMultiplex`,
```
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/codegen/recipes/ecr.go]: Great catch! I had the same idea in the beginning then second guessed myself...

File: plugins/source/aws/codegen/recipes/ecr.go [hunk 3]
```diff
@@ -1,13 +1,51 @@
...
 
 func ECRResources() []*Resource {
 	resources := []*Resource{
+		{
+			SubService: "registries",
+			Struct:     &ecr.DescribeRegistryOutput{},
+			SkipFields: []string{"RegistryId", "ResultMetadata"},
+			Multiplex:  `client.ServiceAccountRegionMultiplexer("api.ecr")`,
+			ExtraColumns: append(
+				defaultRegionalColumns,
```
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/codegen/recipes/ecr.go]: Done. Added `defaultRegionalColumnsPK` that adds the PK attribute

**Review by hermanschaaf — APPROVED** [review]
Nice 👍

**Pull request merged**